Ui4 server settings

velissarious opened the pull request.

velissarious committed:
feat: Edit RunnerSettings by JavaFX forms



Add local an HPC settings screen

kozusznik commented on src/main/java/cz/it4i/parallel/ui/HPCSettingsScreenController.java:
```diff
@@ -0,0 +1,225 @@
+public class HPCSettingsScreenController extends AnchorPane {
+
+	@FXML
+	public TextField hostTextField;
```
Consider make it private (as others)

velissarious replied:
I made all the FXML properties in both "LocalSettingsScreenController" and "HPCSettingsScreenController" private.

kozusznik commented on src/main/java/cz/it4i/parallel/ui/HPCSettingsScreenWindow.java:
```diff
@@ -0,0 +1,92 @@
+		parentStage.showAndWait();
+	}
+
+	private void setInitialTextFieldText() {
```
Consider its moving into ...Controller

velissarious replied:
I move this method and the equivalent method of the "LocalSettingsScreenWindow" in their controllers.

kozusznik commented on src/main/java/cz/it4i/parallel/ui/HPCSettingsScreenWindow.java:
```diff
@@ -0,0 +1,92 @@
+
+	private HPCSettingsScreenController controller;
+
+	private HPCSettings settings;
```
Settings previously used for same window(in case that profile does not contain settings) should be probably stored with PrefService.

velissarious replied:
Settings used for the same window are now stored with PrefService using the helper class "LastFormLoader".

kozusznik commented on src/main/java/cz/it4i/parallel/runners/HPCSettings.java:
```diff
@@ -25,7 +25,7 @@
 	
 	private String authenticationChoice;
 	
-	private transient String password;
```
It needs to be transient.

velissarious replied:
I made the password transient in HPCSettings.

kozusznik reviewed (commented):
Please consider comments.

velissarious committed:
feat: Persist local settings last form storage

Use reasonable default values in settings forms.

kozusznik committed:
code: Use pure JavaFX forms for paradigm manager

Make HPCSettingsWindow fully modal with parent.

Remove using FXFrame as base class for JavaFX windows.

chore: use swing-javafx-ui:0.1.0

kozusznik commented on src/main/java/cz/it4i/parallel/ui/LastFormLoader.java:
```diff
@@ -0,0 +1,76 @@
+
+//Code responsible for loading and storing the last used form values:
+@Slf4j
+public class LastFormLoader {
```
Consider package visibility.

velissarious replied:
LastFormLoader has now package visibility.

kozusznik commented on src/main/java/cz/it4i/parallel/ui/LastFormLoader.java:
```diff
@@ -0,0 +1,76 @@
+	
+	private String formStorage;
+	
+	@Parameter
```
@Parameter is not required - it is set in constructor.

velissarious replied:
Removed the @Parameter.

kozusznik commented on src/main/java/cz/it4i/parallel/ui/LastFormLoader.java:
```diff
@@ -0,0 +1,76 @@
+
+	public void storeLastForm(Object form) {
+		String serializedForm = serializeForm(form);
+		prefService.put(this.getClass(), formStorage, serializedForm);
```
Use a class of the specific form as the parameter(instead of this.getClass() ) in a calling of the method put - it will be passed to the constructor.

velissarious replied:
Added Class as a parameter in LastFormLoader constructor and the correct window class for each form is now used.

kozusznik commented on src/main/java/cz/it4i/parallel/ui/LastFormLoader.java:
```diff
@@ -0,0 +1,76 @@
+
+	private String serializeForm(final Object form) {
+		try {
+			final ByteArrayOutputStream baos = new ByteArrayOutputStream();
```
Consider construction during construction of ObjectOutputStream

velissarious replied:
The new ByteArrayOutputStream is used later and it is not possible to construct it during construction of ObjectOutputStream.

kozusznik commented on src/main/java/cz/it4i/parallel/ui/LastFormLoader.java:
```diff
@@ -0,0 +1,76 @@
+		prefService.put(this.getClass(), formStorage, serializedForm);
+	}
+
+	public Object loadLastForm() {
```
It could return directly subtype of RunnerSettings - LastFormLoader will be generic type.

velissarious replied:
Changed Object to RunnerSettings in LastFormLoader.

kozusznik replied:
I meant that LastFormLoader will be generic type - look at the commit https://github.com/fiji-hpc/scijava-parallel/commit/db06de467fdc99cdbd869b22490fada262dc0eca .

kozusznik commented on src/main/java/cz/it4i/parallel/ui/LastFormLoader.java:
```diff
@@ -0,0 +1,76 @@
+			return deserializeForm(tempString);
+		}
+		catch (Exception exc) {
+			log.info(exc.getMessage());
```
Probably log it as error or warn

velissarious replied:
It is now logged as an error.

kozusznik reviewed (approved):
I have made additional comments.
